### PR TITLE
basic ShellyPlug-S support

### PR DESCRIPTION
**Pull request by eurich**

### CODE REVIEW

**Comment by eurich** [comment]
Important: Revert back to stock firmware fails .. I'll try to debug and revover the device..

**Comment by kzyapkov** [comment]
file `self_test` must be preserved, remove https://github.com/mongoose-os-apps/shelly-homekit/blob/master/src/shelly_main.c#L342

**Comment by eurich** [comment]
thank you @kzyapkov .. reopend..

**Comment by andyblac** [comment]
@eurich what about my comment? line 116, why change it ?

File: mos.yml [hunk 1]
```diff
@@ -113,7 +113,36 @@ conds:
         - ["sw1.id", 0]
         - ["sw1.name", "Shelly SW"]
         - ["sw1.out_gpio", 15]
-        - ["sw1.in_gpio", 4]
+        - ["sw1.in_gpio", 5]
```
[the remark this answers is not in the document]
Comment (eurich) [reply mos.yml]: @andyblac sorry, was a mistake and not wanted.. updated the PR.

[the remark this answers is not in the document]
Comment (andyblac) [reply mos.yml]: ok, np, just thought I would double check in case my commit had an issue.

File: mos.yml [hunk 2]
```diff
@@ -114,6 +114,35 @@ conds:
         - ["sw1.name", "Shelly SW"]
         - ["sw1.out_gpio", 15]
         - ["sw1.in_gpio", 4]
+  - when: build_vars.MODEL == "ShellyPlugS"
```
Comment (rojer) [inline mos.yml]: please insert a blank line before this

File: mos.yml [hunk 3]
```diff
@@ -114,6 +114,35 @@ conds:
...
+        MGOS_MBEDTLS_ENABLE_ATCA: 0
+      cdefs:
+        LED_GPIO: 0
+        LED_ON: 0
+        PRODUCT_MODEL: '"ShellyPlugS"'
+        PRODUCT_HW_REV: '"1.0"'
+        NUM_SWITCHES: 1
+        MG_ENABLE_SSL: 0
+      config_schema:
+        - ["device.id", "shellyplug-s-??????"]
```
Comment (rojer) [inline mos.yml]: does this match the stock fw?

Comment (eurich) [reply mos.yml]: @rojer yes, I think so..  after factory reset (stock firmware) the wifi-AP name is shellyplug-s-7A0920..

**Comment by eurich** [comment]
removed the lin @kzyapkov mentioned.. reverting back to stock fw is working fine now  👍

**Comment by rojer** [comment]
built and pushed a release

**Pull request merged**